internal/util.go

```diff
@@ -0,0 +1,402 @@
+// Package internal contains the utility functions used across the codebase.
+package internal
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/kinvolk/fanotify-poc/pkg/containerd"
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+	"github.com/s3rj1k/go-fanotify/fanotify"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type ContainerNotifier struct {
+	NotifyFD   *fanotify.NotifyFD
+	cnt        *containerd.Container
+	firstEvent bool
+	sha256Sums map[string]string
+	rootFSPath string
+}
+
+const (
+	podKey   = "enforce.k8s.io"
+	podValue = "deny-third-party-execution"
+)
+
+const (
+	RuntimeDocker     = "docker"
+	runtimeContainerd = "containerd"
+)
+
+var containerdNamespace string
+
+func CheckRuntime(hostRuntime string) {
```

Comment:
this function is a bit misleading. It checks runtime but also modified the containerd namespace.